### PR TITLE
CONSOLE-4124: replace deprecated components in ResourceDropdown

### PR DESCRIPTION
**Pull request by rhamilto**
**Note:**  [CONSOLE-4124](https://issues.redhat.com//browse/CONSOLE-4124) will be addressed by 4 PRs, of which this is 1.

After

Events

https://github.com/user-attachments/assets/683216f9-4115-4121-8c4d-ea9420d3ec3a

Search

https://github.com/user-attachments/assets/05049e46-b421-4ffd-a751-9f85db793d48

### CODE REVIEW

File: frontend/public/components/resource-dropdown.tsx [hunk 1]
```diff
@@ -198,58 +245,217 @@ const ResourceListDropdown_: React.SFC<ResourceListDropdownProps> = (props) => {
...
+    <div className={className}>
+      <Select
+        toggle={toggle}
+        onSelect={(_event, selection) => onSelect(selection as string)}
+        selected={selected}
+        isOpen={isOpen}
+        onOpenChange={(open) => {
+          !open && closeMenu();
+        }}
+        // maxMenuHeight="60vh" A bug in the current PatternFly version prevents usage
```
Comment (rhamilto) [inline frontend/public/components/resource-dropdown.tsx:449]: @sg00dwin, have you run in to this yet?  I'm assuming it [has been fixed](https://github.com/patternfly/patternfly-react/issues?q=is%3Aissue+is%3Aclosed+maxMenuHeight), we're just not using a version that includes the fix.

Comment (sg00dwin) [reply frontend/public/components/resource-dropdown.tsx:455]: @rhamilto I hadn't hit that yet. But based on there docs it should work.

**Comment by sg00dwin** [comment]
@rhamilto The keyboard navigation isn't working when trying to move from the Select to the Select Options.  Note: I opened a related keyboard nav issue upstream, in the case of trying to select a checkbox using the enter key. https://github.com/patternfly/patternfly-react/issues/10813

**Comment by sg00dwin** [comment]
The SelectOption and it's resource badge are not in alignment with the checkbox.
Screenshot 2024-08-13 at 3 05 16 PM
Screenshot 2024-08-13 at 2 54 17 PM

**Comment by rhamilto** [comment]
> The SelectOption and it's resource badge are no longer in alignment with the checkbox. Screenshot 2024-08-13 at 3 05 16 PM Screenshot 2024-08-13 at 2 54 17 PM

Good catch, @sg00dwin.  This looks like the default behavior in PatternFly.  We could override, but i'm not sure we should.  WDYT?

Screenshot 2024-08-14 at 11 59 34 AM

**Comment by sg00dwin** [comment]
> > The SelectOption and it's resource badge are no longer in alignment with the checkbox. Screenshot 2024-08-13 at 3 05 16 PM Screenshot 2024-08-13 at 2 54 17 PM
> 
> Good catch, @sg00dwin. This looks like the default behavior in PatternFly. We could override, but i'm not sure we should. WDYT?
> 
What caught my eye and looked wonky were the checkbox and resource badges being misaligned on some. I'm fine either way.

**Comment by rhamilto** [comment]
Tech debt so manually adding labels:
/label px-approved
/label docs-approved

/assign @yapei
/assign @sg00dwin

File: frontend/public/components/resource-dropdown.tsx (hunk 1 above)
Comment (rhamilto) [reply frontend/public/components/resource-dropdown.tsx]: Added CSS to address so we retain existing behavior.

**Comment by rhamilto** [comment]
@sg00dwin, I believe the keyboard navigation issue is fixed.  PTAL.

**Comment by yapei** [comment]
It looks like we have too much margin in resource dropdown
PR-resource-type-selection

for comparison(cluster without PR changes)
Screenshot 2024-08-20 at 3 15 51 PM

**Comment by yapei** [comment]
> @sg00dwin, I believe the keyboard navigation issue is fixed. PTAL.

@rhamilto  navigating through resources with `tab` key doesn't work though

**Comment by rhamilto** [comment]
> It looks like we have too much margin in resource dropdown PR-resource-type-selection
> 
> for comparison(cluster without PR changes) Screenshot 2024-08-20 at 3 15 51 PM

@yapei, this is expected given the differences in the layout of the components.

**Comment by rhamilto** [comment]
> > @sg00dwin, I believe the keyboard navigation issue is fixed. PTAL.
> 
> @rhamilto navigating through resources with `tab` key doesn't work though

This is expected behavior.  See https://www.patternfly.org/components/menus/select#multiple-typeahead-with-checkboxes

**Comment by yapei** [comment]
Thanks for confirmation @rhamilto 
/label qe-approved

**Pull request merged**